### ISSUE TITLE
Incompatibility with Charm: Getting stuck in portals after the new dimension loads.

### ISSUE THREAD

**Issue by PkunkArilou** [LudoCrypt/Music-Duration-Reducer#2]
_MC 1.16.5_
_MDR 1.2.0_
_Charm-fabric-1.16.5-2.3.2_
_https://www.curseforge.com/minecraft/mc-mods/charm_

_Issue on the Charm board here:_
_https://github.com/svenhjol/Charm/issues/548_

Testing Charm with MDR, I get stuck in portals; usually on the first try. The new dimension loads but the character can't move and isn't visible in 3rd person view.

Testing Charm without MDR, I never get stuck in a portal.  

The log in the issue linked above points to a line in the Charm source that's trying to send a non-positive parameter to Random.nextInt(). This is where I'm getting the problem also. It appears that without MDR, that parameter is never non-positive, but with MDR it often is, due to MDR's **musicDr_getMinDelay** I imagine.

To reproduce:
* Load the 2 mods (versions above), and Fabric API 0.34.7+1.16.
* Make a new world.
* Creative mode, build a nether portal and move through it.
* If not already stuck, try returning through the same portal, or making a new portal and trying again. It usually occurs on the first attempt, and I've never had to try more than 3 attempts to produce the issue.

WORKAROUND
I think zeroed minDelays are probably something Charm needs to handle, but for now you can go to MDR's config and set Min Delay to anything greater than 0. I only tried a handful of times but that appears to get around the issue.

_(Edit -- Forgot to mention the Fabric API mod is also required to start the client.)_

**Comment by LudoCrypt** [comment]
Unable to reproduce. I even set both of min and max delay to 0 and still no bug.

**Comment by PkunkArilou** [comment]
That's interesting. I made a new instance to check again and made some new worlds in it.
* First world: Occurred with 4th built portal
* Second world: Occurred with 2nd built portal
* Third world: Occurred with 1st use of **/execute in minecraft:the_nether run teleport ~ ~ ~**
* Fourth world: Did not occur with 1st use of /execute in minecraft:the_nether run teleport ~ ~ ~, but occurred with 1st use of **/execute in minecraft:overworld run teleport ~ ~ ~** (coming from the nether).

Possible reason for our differences: I think I had the wrong impression about re-trying existing portals. Tonight I found that existing portals caused the problem much, much less than newly-built portals; re-using an existing portal took me 11 attempts before producing the bug. So my bad if that's what you were testing with. 

I'd be surprised to find it ultimately un-reproduceable though, simply because Charm is allowing a zero to be passed to Random.nextInt(), which gives an IllegalArgumentException.

Anyway, that's not MDR code so no complaints with closing this; just reporting findings. Anyone coming across this problem should keep an eye on the Charm page, and use the above workaround.
